src/test/java/com/training/web/tests/flipkart/FlipkartTest.java

```diff
@@ -67,4 +73,94 @@ public void test() throws WebDriverException {
          */
         driver.close();
     }
+
+    @Test
+    public void testCartAddition() {
+        Integer[] products = {2, 3};
+
+        /**
+         * System Property for Chrome Driver
+         */
+        System.setProperty("webdriver.chrome.driver", "package/resources/chromedriver.exe");
+
+        /**
+         * Instantiate a ChromeDriver class.
+         */
+        ChromeOptions options = new ChromeOptions().addArguments("incognito");
+        DesiredCapabilities capabilities = DesiredCapabilities.chrome();
+        capabilities.setCapability(ChromeOptions.CAPABILITY, options);
+        WebDriver driver = new ChromeDriver(capabilities);
+        WebDriverWait wait = new WebDriverWait(driver, 30);
+
+
+        /**
+         * Launch Website and maximise
+         */
+        driver.navigate().to("https://www.flipkart.com/");
+        driver.manage().timeouts().implicitlyWait(10, TimeUnit.SECONDS);
+        driver.manage().window().maximize();
+        String parentWindow = driver.getWindowHandle();
+
+        /**
+         * Close popup and search for shoes
+         */
+        ResultPage resultPage = new FlipkartHomePage(driver).closePopup().sendKeysToSearchBox("shoes")
+                .clickSearch().sortLowToHigh();
+
+        /**
+         * Select and add to cart item 2 and 3
+         */
+        ArrayList<String> productNames = new ArrayList<>();
+        ArrayList<Integer> priceList = new ArrayList<>();
+        ProductPage productPage = new ProductPage(driver);
+        resultPage.getProductsList();
+
+        for (int i : products) {
+            resultPage.clickProduct(i);
+            Set<String> s = driver.getWindowHandles();
+            String child_window = null;
+            Iterator<String> I1 = s.iterator();
+            while (I1.hasNext()) {
+                if (I1.next() != parentWindow) ;
+                child_window = I1.next();
+            }
+            driver.switchTo().window(child_window);
+            productPage.clickSize();
+            productNames.add(productPage.productName());
+            priceList.add(productPage.productPrice());
+            productPage.addToCart();
+            driver.close();
+            driver.switchTo().window(parentWindow);
+        }
+        resultPage.goToCart();
+
+        /**
+         * Asserting for right products
+         */
+        CartPage cartPage = new CartPage(driver);
```

Comment:
As we discussed earler, please do not use : `Classname classname = new Classname()` anywhere in your code